pkg/providers/instance/instance.go

```diff
@@ -452,26 +453,25 @@ func isSKUNotAvailable(err error) bool {
 }
 
 func (p *Provider) handleResponseErrors(ctx context.Context, instanceType *corecloudprovider.InstanceType, zone, capacityType string, err error) error {
-	if sdkerrors.SubscriptionQuotaHasBeenReached(err) {
-		// Subscription quota reached, mark the instance type as unavailable in all zones available to the offering
+	if sdkerrors.SKUFamilyQuotaHasBeenReached(err) {
+		// Subscription quota has reached for this vm sku, mark the instance type as unavailable in all zones available to the offering
 		// This will also update the TTL for an existing offering in the cache that is already unavailable
+
+		logging.FromContext(ctx).Error(err)
 		for _, offering := range instanceType.Offerings {
 			if offering.CapacityType != capacityType {
 				continue
-			}
-
-			// If we have a quota limit of 0 vcpus, we mark the offerings unavailable for an hour.
+			} // If we have a quota limit of 0 vcpus, we mark the offerings unavailable for an hour.
 			// CPU limits of 0 are usually due to a subscription having no allocated quota for that instance type at all on the subscription.
 			if cpuLimitIsZero(err) {
 				p.unavailableOfferings.MarkUnavailableWithTTL(ctx, SubscriptionQuotaReachedReason, instanceType.Name, offering.Zone, capacityType, SubscriptionQuotaReachedTTL)
 			} else {
 				p.unavailableOfferings.MarkUnavailable(ctx, SubscriptionQuotaReachedReason, instanceType.Name, offering.Zone, capacityType)
 			}
 		}
-	} else if sdkerrors.ZonalAllocationFailureOccurred(err) {
-		p.unavailableOfferings.MarkUnavailable(ctx, ZonalAllocationFailureReason, instanceType.Name, zone, corev1beta1.CapacityTypeOnDemand)
-		p.unavailableOfferings.MarkUnavailable(ctx, ZonalAllocationFailureReason, instanceType.Name, zone, corev1beta1.CapacityTypeSpot)
-	} else if isSKUNotAvailable(err) {
+		return fmt.Errorf("subscription level quota for %s has been reached (may try provision an alternative instance type)", instanceType.Name)
+	}
+	if isSKUNotAvailable(err) {
```

Comment:
Wrapping the original error is a better pattern, I think, but if the conclusion is it makes for unacceptable status error UX, let's be consistent and replace it with logging in this codepath as well.